aspnetcore/signalr/javascript-client.md

```diff
@@ -0,0 +1,108 @@
+---
+title: ASP.NET Core SignalR JavaScript client
+author: rachelappel
+description: Create a JavaScript client for an ASP.NET Core SignalR hub.
+manager: wpickett
+ms.author: rachelap
+ms.custom: mvc
+ms.date: 04/06/2018
+ms.prod: aspnet-core
+ms.technology: aspnet
+ms.topic: article
+uid: signalr/javascript-client
+---
+
+# ASP.NET Core SignalR JavaScript client
+
+By [Rachel Appel](http://twitter.com/rachelappel)
+
+The ASP.NET Core SignalR JavaScript client library enables developers to call server-side hub code.
+
+## Install the SignalR client package
+
+The SignalR JavaScript client library is delivered as an [npm](https://www.npmjs.com/) package. If you're using Visual Studio, call `npm install` from the command line in any folder. For Visual Studio Code, call the command from the **Integrated Terminal**.
+
+  ```console
+   npm install @aspnet/signalr
+  ```
+
+Npm installs the package contents in the *node_modules\\@aspnet\signalr\dist\browser* folder. Copy the *signalr.js* or its minified version to a folder in your project.
+
+## Use the SignalR JavaScript client
+
+To use the SignalR JavaScript client, a script reference is required. Since the reference is an HTML `<script>` element, it can go in any kind of client-side page, such as a [Razor Page](xref:mvc/razor-pages/index?tabs=visual-studio), [MVC view](xref:mvc/views/overview), or HTML page.
+
+```html
+<script src="~/scripts/signalr.js"></script>
+```
+
+## Connect to a hub
+
+To connect to a SignalR hub, create a new instance of a connection with the route as its argument. The route maps to a hub with the same name. However, the hub's name can be mixed case, such as `ChatHub`.
+
+```javascript
+const connection = new signalR.HubConnection('/chathub');
+```
+
+While the preceding code sample creates a connection to a hub named `ChatHub`, it doesn't start it. To start the connection, call the `connection.start` method.
+
+```javascript
+connection.start().catch(err => logError(err));
+```
+
+### Specify the connection's transport type
+
+Transports define how servers and clients communicate. To set the transport type when you create the connection by setting the `signalR.TransportType` to *WebSockets*, *LongPolling*, or *ServerSentEvents*. The default, recommended transport is *WebSockets*.
```

Comment:
This has a few problem. Can you show more code? `transportType` is not shown, I think you have the range wrong?

Something like, the following code sets the trans type to WebSockets

Then below tell them the types they can set.

I think I'd combine this with the previous snippet.  better to show code and explain it.

Reply:
To be clear, the default, recommended pattern is to **not provide the transport type at all**. It is not required and by default we will automatically choose an appropriate transport.